pkg/server/server.go

```diff
@@ -100,13 +127,31 @@ func (s *Server) Run() (err error) {
 			continue
 		}
 
-		s.log.Debug("Initializing " + service.Name)
-
+		if cfg != nil {
+			if httpS, ok := service.Instance.(*api.HTTPServer); ok {
```

Comment:
feels a little bit hacky but I guess its ok. Perhaps you can add some comments here explaining this?

Reply:
Added comment.